tests/e2e/utils/service_utils.go

```diff
@@ -100,20 +102,29 @@ func WaitServiceExposureAndGetIPs(cs clientset.Interface, namespace string, name
 }
 
 // WaitServiceExposureAndValidateConnectivity returns IPs of the service and check the connectivity if they are public IPs.
+// Service should have been created before calling this function.
 func WaitServiceExposureAndValidateConnectivity(cs clientset.Interface, ipFamily IPFamily, namespace string, name string, targetIPs []string) ([]string, error) {
 	var service *v1.Service
 	var err error
 	var ips []string
 
+	deleteService := func() {
```

Comment:
how about invoke delete service in deferred funcions
```go
defer func() {
  if err != nil {
	err := DeleteService(cs, namespace, name)
	Expect(err).NotTo(HaveOccurred())
   }
}
```

Reply:
we need to name the returned error as err

Reply:
another approach would be errorgroup

Reply:
Thank you. Updated.